src/otx/engine/engine.py

```diff
@@ -878,6 +882,14 @@ def _build_trainer(self, **kwargs) -> None:
         """Instantiate the trainer based on the model parameters."""
         if self._cache.requires_update(**kwargs) or self._trainer is None:
             self._cache.update(**kwargs)
+            # set up xpu device
+            if self._device.accelerator == DeviceType.xpu:
+                self._cache.update(strategy="xpu_single")
+                # add plugin for Automatic Mixed Precision on XPU
+                if self._cache.args["precision"] == 16:
```

Comment:
```suggestion
                if self._cache.args.get("precision", 32) == 16:
```
Some models don't have precision, which can make an error.

Reply:
Done https://github.com/openvinotoolkit/training_extensions/pull/3295/commits/023cc8f51a90a4c20329bdc67a7c3a3310db5936